samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,399 @@
+
+To create a **descriptor set**, we need to allocate it from a **descriptor set pool** and give it a specific **descriptor set layout**.
+
+After a **descriptor set** is allocated, it needs to be updated with the **descriptors**. The update process requires us to specify a list of **write operations**, where a write operation is a [`VkWriteDescriptorSet`](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkWriteDescriptorSet.html) struct.
+
+Then the valid **descriptor set** is bound to a command buffer so that when `vkCmdDraw*()` commands are run, the right resources are made available in the GPU.
+
+#### *Buffer Object*
+
+For all the descriptor set sections below, we will use one such resource known as a **buffer object**, as this will be what we use to store our MVP data.
+
+A **buffer object** in Vulkan is a type of `VkBuffer`, created with the respective buffer usage flag. For uniform buffer objects we use the `VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT` flag, and for shader storage buffer objects we use the `VK_BUFFER_USAGE_STORAGE_BUFFER_BIT` flag. These map to the shader `<storage>` type `uniform`s and `buffer`s respectively. 
+
+### **Performance**
+
+![single ubo](images/descriptor_set_performance.jpg)
+
+While it is not straightforward to perform a 1:1 comparison between push constants and descriptor sets, the sample does show static descriptor sets outperforming push constants.
+
+When we compare the frametime with [push constants](#push-constants) we can see it is the same (16.7ms), however it is the load/store cycles we want to look at. They drop from 266 k/s to 123 k/s, showing that the GPU is worked more in the case of push constants to achieve the same visual results.
+
+## **Dynamic Descriptor Sets**
+
+### **Introduction**
+
+Dynamic descriptor sets differ to the regular descriptor sets because they allow an offset to be specified when we aere *binding* (`vkCmdBindDescriptorSets`) the descriptor set. This dynamic offset can be used in addition to the base offset used at the time of updating the descriptor set. 
+
+One case in which this can be useful is:
+
+1. Allocating one giant **uniform buffer object** containing all the world matrices of the meshes in your scene. 
+2. Allocating a **descriptor set** with a binding containing the `VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC` flag that then points to the **UBO** you just created.
+3. In our draw call, for each mesh, *dynamically* offset into the giant **uniform buffer object**.
+
+### **Performance**
+
+![dynamic ubo](images/dynamic_descriptor_set_performance.jpg)
+
+In the screenshot above we can see the load/store cycles stay roughly around the same compared to [static uniform buffer objects](#uniform-buffer-objects). However, the frame time goes up from 16.7 ms to 20.9 ms. This is due to the extra time you need to spend every frame determining the dynamic offsets, that you need to send in the bind call (`vkCmdBindDescriptorSets`).
```

Comment:
Arm specific numbers

Reply:
How would you like to approach the quoting of numbers? Shall I take readings from a few different vendors?

Reply:
I think we just need to make it clear that these are example figures on Arm GPUs. I think that's what happened in the other samples.